orchestrator_mapreduce.py

```
@@ -0,0 +1,478 @@
+
+        protocol = load_protocol(protocol_name)
+        if not protocol:
+            raise Exception(f"Protocol '{protocol_name}' not found")
+
+        task_fn = protocol["task"]
+        loop = asyncio.get_running_loop()
+
+        # Pass inputs to the protocol task function.
+        # Protocols that accept kwargs will receive them;
+        # legacy protocols with no parameters still work via fallback.
+        def _execute_with_inputs():
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_execute_with_inputs`

(ANN202)